internal/rfc7512/rfc7512.go

```diff
@@ -0,0 +1,182 @@
+package rfc7512
```

Comment:
Would `package hsm` work? And package docs

```
// Package hsm implements blah.
//
// More context if you want to give it. For more information, see this and that standard.
```

Reply:
Could possibly also change PKCS11URI -> URI, LoadPKCS11Key -> LoadKey

Reply:
I added package docs, but this isn't really an `hsm` package. Maybe `pkcs11uri` or `pkcs11-uri` but I honestly prefer `rfc7512`.
For struct and function names I'm following `crypto11`'s example for the most part.

Reply:
Stop 👏deferring 👏to 👏the 👏standards. Think of it like that book, The Giver: You read the standards so we don't have to.

Reference on names: https://blog.golang.org/package-names. This package lets you parse HSM URIs, it lets you 'connect' to an HSM. If this isn't an HSM package, what is a simple noun that describes it? The package docs are non-sensical; it'd be good to move the stuff in LoadPKCS11Signer's docs up here, with even more elaboration.

Reply:
I much prefer @Bren2010's suggested naming conventions, and agree that your fellow programmers would be better suited with a short layman's description instead of just referencing the RFC. The Giver analogy is excellent.

That said, this is an internal package with pretty limited scope, so I'm not too stressed either way.